core/src/test/scala/org/bykn/bosatsu/pattern/StrPartTest.scala

```diff
@@ -0,0 +1,103 @@
+package org.bykn.bosatsu.pattern
+
+import org.bykn.bosatsu.Generators.{genStrPat, genValidUtf}
+import org.bykn.bosatsu.StringUtil
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Gen
+
+class StrPartTest extends munit.ScalaCheckSuite {
```

Comment:
this test actually also tests that all the matching algorithms agree. It seems we have something like 4 of them:

1. StrPart
2. SeqPattern + matcher
3. NamedSeqPattern + matcher
4. java regex

Reply:
I *think* StrPart is the fastest, but I haven't carefully benchmarked it, but they should all agree, and the speed only matters for the MatchlessToValue runtime, in normalization, it doesn't matter I don't think.